tests/e2e/spawn/spawn.go

```diff
@@ -72,3 +76,39 @@ func Command(command string, arguments ...string) (string, error) {
 
 	return string(outBytes), err
 }
+
+// CommandExecWithContext runs a command with its arguments, kills the command after context is done
+// and returns the combined stdout, stderr or the error.
+func CommandExecWithContext(ctx context.Context, command string, arguments ...string) (string, error) {
+	cmd := exec.Command(command, arguments...)
+	var b bytes.Buffer
+	cmd.Stdout = &b
+	cmd.Stderr = &b
+
+	err := cmd.Start()
+	if err != nil {
+		return "", fmt.Errorf("error starting command : %w", err)
+	}
+
+	waitErrChan := make(chan error, 1)
+	go func(errChan chan error) {
```

Comment:
is errChan needed here ?

Reply:
it is ideally better to have a go routine sharing chan and not direct vars

Reply:
it looks like it is unused in the go routine ?

Reply:
nope it is being used in one of the E2E. When the command exits with error, this is being used.